pkg/submitter/composite_submitter.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 Preferred Networks, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package submitter
+
+import (
+	"github.com/containerd/containerd/log"
+	"k8s.io/kubernetes/pkg/scheduler/algorithm"
+
+	"github.com/pfnet-research/k8s-cluster-simulator/pkg/clock"
+	"github.com/pfnet-research/k8s-cluster-simulator/pkg/metrics"
+)
+
+// CompositeSubmitter gathers multiple submitters into a single composite submitter.
+type CompositeSubmitter struct {
+	submitters map[string]Submitter
+}
+
+// NewCompositeSubmitter creates a new CompositeSubmitter with the given submitters.
+// Submitters are identified by their name of string.
+func NewCompositeSubmitter(submitters map[string]Submitter) *CompositeSubmitter {
+	return &CompositeSubmitter{
+		submitters,
```

Comment:
I think we need to deep-copy the map object, otherwise, changes in the caller affect the logic of this object.

Reply:
I fixed `NewCompositeSubmitter` to deep-copy the map.

However, since `Submitter` is an interface, the underlying submitters can still be accessed from outside of `CompositeSubmitter`. We *could* deep-copy interfaces, but it seems to me, a golang newbie, that we will need to take [a dirty way](https://gist.github.com/soroushjp/0ec92102641ddfc3ad5515ca76405f4d) or require the interfaces to have a method like `Cloneable`. 
I personally don't mind leaving this behavior (= underlying submitters are accessible) as a specification.

I wish golang had an ownership model like Rust :man_shrugging: